### ISSUE TITLE
BatchNorm2d when batch size 1 works, what is it doing?

### ISSUE THREAD

**Issue by mikolchon** [pytorch/pytorch#6468]
`BatchNorm2d` works even when batch size is 1, which puzzles me. So what is it doing when batch size is 1? The only related thread I could find is https://github.com/pytorch/pytorch/issues/1381 without much explanation.

minimal example:
```
x = Variable(torch.randn(1,2,3,3))
m = nn.BatchNorm2d(2)
y = m(x)
```

**Comment by SsnL** [comment]
It' is doing essentially instance norm.

**Comment by apaszke** [comment]
We're using GitHub issues for bug reports only, and all questions should be posted on [our forums](http://discuss.pytorch.org).

**Comment by SsnL** [comment]
That is normalizing `[B x C x *]` over the dimensions `[*]`

**Comment by mikolchon** [comment]
Thanks @SsnL, I will repost the question in the proper forum just in case someone else needs to know.

**Comment by SsnL** [comment]
resolved on forum
